ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/gui/BoardPanel.java

```diff
@@ -50,17 +52,36 @@ public BoardPanel(BoardSize boardSize, int boardIndex, GlassPane globalGlassPane
         };
 
         setOpaque(false);
-        setLayout(new GridLayout(boardSize.width + 1, boardSize.height));
-        setPreferredSize(new Dimension((boardSize.width + 1) * 48, (boardSize.height + 1) * 48));
-        createGrid(boardSize);
+        setLayout(new GridLayout(mBoardSize.width + 1, mBoardSize.height));
+        setPreferredSize(new Dimension((mBoardSize.width + 1) * 48, (mBoardSize.height + 1) * 48));
+        createGrid();
     }
 
-    private void createGrid(BoardSize boardSize) {
-        for (int y = boardSize.height; y > 0; y--) {
+    public void rescaleBoard(int panelWidth, int panelHeight){
+        Double boardScale = (panelHeight * .85);
+        Double scale = (panelHeight * 1.0) / (panelWidth * 1.0);
+        if(scale > .75) {
+            boardScale = panelHeight * (.80 - (scale - .75));
+        }
+        boardScale = boardScale / mBoardSize.width + 1;
+
+        if(boardScale > 0) {
+            setPreferredSize(new Dimension((mBoardSize.width + 1) * boardScale.intValue(), (mBoardSize.height + 1) * boardScale.intValue()));
```

Comment:
if you have a lowercase `double`, you can cast to int with `(int)` instead of using `.intValue()`